### PR TITLE
Autreg 131 javascript linting

### PR DESCRIPTION
**Pull request by Eiriknil**

### CODE REVIEW

File: cypress/support/step_definitions/anonymous_user_view_entity_registry_metadata_HTML.js [hunk 1]
```diff
@@ -18,27 +18,35 @@
...
 
-then('they see metadata related to the entity registry regarding:', (dataTable) =>{
-	const attributeArray = dataTable.rawTable;
+  cy.get('@registryName').then((registryName) => {
+    const createRegistryEndpoint = '/registry/' + registryName;
+    cy.request({
+      url: createRegistryEndpoint,
+      method: 'GET',
+      headers: {
+        'content-type': 'text/html',
```
Comment (axthosarouris) [inline cypress/support/step_definitions/anonymous_user_view_entity_registry_metadata_HTML.js]: A const TEXT_HTML is defined in another file. Better to use a const here too? Or define some global constants?

Comment (axthosarouris) [reply cypress/support/step_definitions/anonymous_user_view_entity_registry_metadata_HTML.js]: Also do we by content-type mean 'accept'?

File: cypress/support/step_definitions/anonymous_user_view_entity_registry_metadata_RDF.js [hunk 2]
```diff
@@ -11,31 +11,26 @@
...
-			cy.wrap(response).as('registryMetadata');
-		})
-	})
+  cy.get('@registryName').then((registryName) => {
+    const createRegistryEndpoint = '/registry/' + registryName;
+    cy.request({
+      url: createRegistryEndpoint,
+      method: 'GET',
+      headers: {
+        'content-type': 'text/html',
```
Comment (axthosarouris) [inline cypress/support/step_definitions/anonymous_user_view_entity_registry_metadata_RDF.js]: see earlier comment

File: cypress/support/step_definitions/api_admin_delete_empty_registry.js [hunk 3]
```diff
@@ -1,44 +1,45 @@
...
+  // delete empty registry
+  cy.get('@registryName').then((registryName) => {
+    cy.get('@registryAdminApiKey').then((apiKey) => {
+      let url = '/registry/' + registryName;
+      cy.request({
+        url: url,
+        method: 'DELETE',
+        headers: {
+          'api-key': apiKey,
+          'content-type': 'application/json',
```
Comment (axthosarouris) [inline cypress/support/step_definitions/api_admin_delete_empty_registry.js]: 'accept' ?

File: cypress/support/step_definitions/api_admin_update_empty_registry_schema.js [hunk 4]
```diff
@@ -4,58 +4,60 @@
...
+      });
+
+      let registryUpdateUrl = '/registry/' + registryName + '/schema';
+      let updatedSchema = 'updatedTestSchema';
+      cy.request({
+        url: registryUpdateUrl,
+        method: 'PUT',
+        headers: {
+          'api-key': apiKey,
+          'content-type': 'application/json',
```
Comment (axthosarouris) [inline cypress/support/step_definitions/api_admin_update_empty_registry_schema.js]: 'accept' ?

File: cypress/support/step_definitions/api_admin_update_registry_metadata_.js [hunk 5]
```diff
@@ -4,55 +4,57 @@
...
+      cy.fixture('registryTestMetadataUpdated.json').then((updatedSchema) => {
+        updatedSchema.id = registryName;
+        cy.log('description: ' + updatedSchema.metadata.description);
+        cy.request({
+          url: registryUpdateUrl,
+          method: 'PUT',
+          body: updatedSchema,
+          headers: {
+            'api-key': apiKey,
+            'Content-Type': 'application/json',
```
Comment (axthosarouris) [inline cypress/support/step_definitions/api_admin_update_registry_metadata_.js]: If text/html is saved as  const, shouldn't application/json be also saved as const?
Also,  'accept' ?

File: cypress/support/step_definitions/api_admin_user_provides_api_key.js [hunk 6]
```diff
@@ -3,34 +3,36 @@
...
+        .then((testSchema) => {
+          testSchema.id = registryName;
+          let createUrl = '/registry';
+          cy.request({
+            url: createUrl,
+            method: 'POST',
+            body: testSchema,
+            headers: {
+              'api-key': apiAdminApiKey,
+              'content-type': 'application/json',
```
Comment (axthosarouris) [inline cypress/support/step_definitions/api_admin_user_provides_api_key.js]: same as previous comments

File: cypress/support/step_definitions/anonymous_user_view_entity_registry_metadata_HTML.js (hunk 1 above)
Comment (Eiriknil) [reply cypress/support/step_definitions/anonymous_user_view_entity_registry_metadata_HTML.js]: Fixed

File: cypress/support/step_definitions/anonymous_user_view_entity_registry_metadata_RDF.js (hunk 2 above)
Comment (Eiriknil) [reply cypress/support/step_definitions/anonymous_user_view_entity_registry_metadata_RDF.js]: copy-pasta, changed

File: cypress/support/step_definitions/api_admin_update_empty_registry_schema.js (hunk 4 above)
Comment (Eiriknil) [reply cypress/support/step_definitions/api_admin_update_empty_registry_schema.js]: PUT, content-type is correct

File: cypress/support/step_definitions/api_admin_user_provides_api_key.js (hunk 6 above)
Comment (Eiriknil) [reply cypress/support/step_definitions/api_admin_user_provides_api_key.js]: POST, content-type is correct

File: cypress/support/step_definitions/api_admin_update_registry_metadata_.js (hunk 5 above)
Comment (Eiriknil) [reply cypress/support/step_definitions/api_admin_update_registry_metadata_.js]: PUT, content-type is correct

File: cypress/support/step_definitions/api_admin_delete_empty_registry.js (hunk 3 above)
Comment (Eiriknil) [reply cypress/support/step_definitions/api_admin_delete_empty_registry.js]: fixed

File: api/build.gradle [hunk 7]
```diff
@@ -23,6 +23,7 @@ dependencies {
     compile("org.glassfish.jersey.inject:jersey-hk2:2.27") {
         exclude group: 'javax.inject'
     }
+	compile ('com.google.code.gson:gson:2.8.5')
```
Comment (axthosarouris) [inline api/build.gradle]: We can use Jackson

File: api/src/main/java/no/bibsys/web/EntityMessageBodyWriter.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+import no.bibsys.web.model.EntityDto;
+
+@Provider
+@Produces("text/html")
+public class EntityMessageBodyWriter implements MessageBodyWriter<EntityDto> {
+
+    @Override
+    public boolean isWriteable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+
```
Comment (axthosarouris) [inline api/src/main/java/no/bibsys/web/EntityMessageBodyWriter.java]: spaces

File: api/src/main/java/no/bibsys/web/EntityMessageBodyWriter.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Override
+    public void writeTo(EntityDto entity, Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType,
+            MultivaluedMap<String, Object> httpHeaders, OutputStream entityStream)
+            throws IOException, WebApplicationException {
+        
+        try(Writer writer = new PrintWriter(entityStream)){
+        
+            Handlebars handlebars = new Handlebars();
```
Comment (axthosarouris) [inline api/src/main/java/no/bibsys/web/EntityMessageBodyWriter.java]: spaces

File: cypress/support/commands.js [hunk 10]
```diff
@@ -9,131 +9,131 @@
...
+
+  cy.log('api-key = ' + apiKey);
+
+  const url = '/registry/' + registryName;
+  cy.request({
+    url: url,
+    method: 'DELETE',
+    headers: {
+      'api-key': apiKey,
+      'content-type': 'application/json',
```
Comment (axthosarouris) [inline cypress/support/commands.js]: shouldn't this be accept?

File: cypress/support/step_definitions/anonymous_user_view_entity_HTML.js [hunk 11]
```diff
@@ -4,34 +4,36 @@
...
-	cy.get('@registryName').then((registryName) => {
-		cy.get('@entityId').then((entityId) => {
-			const getEntityUrl = '/registry/' + registryName + '/entity/' + entityId;
-			cy.visit(getEntityUrl)
+When(/the anonymous user requests the entity specifying an Accept header with value text\/html/, () => {
+  cy.log('-- anonymous_user_view_entity_HTML.js --')
+  cy.get('@registryName').then((registryName) => {
+    cy.get('@entityId').then((entityId) => {
+      const getEntityUrl = '/registry/' + registryName + '/entity/' + entityId;
+
```
Comment (axthosarouris) [inline cypress/support/step_definitions/anonymous_user_view_entity_HTML.js]: space

File: cypress/support/step_definitions/api_admin_delete_populated_registry.js [hunk 12]
```diff
@@ -4,33 +4,35 @@
...
-})
+    cy.get('@apiAdminApiKey').then((apiKey) => {
+      cy.log('apiKey = ' + apiKey);
+      let url = '/registry/' + registryName;
+      cy.request({
+        url: url,
+        method: 'DELETE',
+        headers: {
+          'api-key': apiKey,
+          'content-type': 'application/json',
```
Comment (axthosarouris) [inline cypress/support/step_definitions/api_admin_delete_populated_registry.js]: accecpt?

File: api/build.gradle (hunk 7 above)
Comment (Eiriknil) [reply api/build.gradle]: Ok

File: cypress/support/commands.js (hunk 10 above)
Comment (Eiriknil) [reply cypress/support/commands.js]: content-type removed

File: cypress/support/step_definitions/api_admin_delete_populated_registry.js (hunk 12 above)
Comment (Eiriknil) [reply cypress/support/step_definitions/api_admin_delete_populated_registry.js]: removed

**Pull request merged**